src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatReader.cs

```diff
@@ -138,7 +138,32 @@ private void ReflectionSetMemberValue(object obj, object memberValue, MemberInfo
                 else
                 {
                     object property = propInfo.GetValue(obj);
-                    MethodInfo addMethod = property.GetType().GetMethod("Add");
+                    Type propertyType = property.GetType();
+                    MethodInfo addMethod = propertyType.GetMethod("Add");
```

Comment:
If there is more than one method called Add you will get an AmbiguosMatchException. I presume that because of the code which follows that this might be possible. There's an overload of GetMethod which takes an array of Type objects to match the parameters. Could this be used to make the code which follows simpler?